### PR TITLE
radio-cli: init at 2.3.1

### LINKED ISSUE #330779: Package request: radio-cli
**Project description**

A simple radio CLI written in rust.

**Metadata**

* homepage URL: https://github.com/margual56/radio-cli
* source URL: https://github.com/margual56/radio-cli
* license: gpl2
* platforms: linux, darwin

### PR DESCRIPTION
**Pull request by luftmensch-luftmensch**
## Description of changes
Closes #330779

[Release notes](https://github.com/margual56/radio-cli/releases/tag/v2.3.1)

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ra/radio-cli/package.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  cargoHash = "sha256-XCzDNUZpyfu4gJr1lUx1/VbLkEv3OnJvYku0eEJbK5s=";
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ openssl ];
+
+  meta = {
+    description = "Simple radio CLI written in rust";
+    homepage = "https://github.com/margual56/radio-cli";
+    changelog = "https://github.com/margual56/radio-cli/releases/tag/v${version}";
+    license = lib.licenses.mit;
```
Comment (IldenH) [inline pkgs/by-name/ra/radio-cli/package.nix:27]:
```suggestion
    license = lib.licenses.gpl2Only;
```

File: pkgs/by-name/ra/radio-cli/package.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  openssl,
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "radio-cli";
+  version = "2.3.1";
+
+  src = fetchFromGitHub {
+    owner = "margual56";
+    repo = "radio-cli";
+    rev = "refs/tags/v${version}";
```
Comment (IldenH) [inline pkgs/by-name/ra/radio-cli/package.nix:15]:
```suggestion
    rev = "v${version}";
```

File: pkgs/by-name/ra/radio-cli/package.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  buildInputs = [ openssl ];
+
+  meta = {
+    description = "Simple radio CLI written in rust";
+    homepage = "https://github.com/margual56/radio-cli";
+    changelog = "https://github.com/margual56/radio-cli/releases/tag/v${version}";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ luftmensch-luftmensch ];
+    mainProgram = "radio-cli";
+  };
```
Comment (IldenH) [inline pkgs/by-name/ra/radio-cli/package.nix:30]:
```suggestion
    platforms = lib.platforms.linux;
  };
```

**Review by IldenH — CHANGES_REQUESTED** [review]

**Review by IldenH — APPROVED** [review]
I don't know why ofborg is failing, but looks good to me.

**Comment by Pandapip1** [comment]
@ofborg eval

**Comment by marcusramberg** [comment]
```
> Select a station to play: lofi
Playing lofi
Info: press 'q' to stop playing this station
thread 'main' panicked at src/main.rs:183:10:
Failed to execute mpv. Is it installed?: Os { code: 2, kind: NotFound, message: "No such file or directory" }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Maybe add a dependency on `mpv`?

**Comment by luftmensch-luftmensch** [comment]
> ```
> > Select a station to play: lofi
> Playing lofi
> Info: press 'q' to stop playing this station
> thread 'main' panicked at src/main.rs:183:10:
> Failed to execute mpv. Is it installed?: Os { code: 2, kind: NotFound, message: "No such file or directory" }
> note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
> ```
> 
> Maybe add a dependency on `mpv`?

Gonna add it later

**Comment by marcusramberg** [comment]
Lgtm!

**Pull request merged**